internal/provider/resource_vdb.go

```diff
@@ -1213,14 +1213,25 @@ func resourceVdbRead(ctx context.Context, d *schema.ResourceData, meta interface
 
 	vdbId := d.Id()
 
-	res, diags := PollForObjectExistence(func() (interface{}, *http.Response, error) {
+	res, diags, statusCode := PollForObjectExistence(func() (interface{}, *http.Response, error) {
 		return client.VDBsApi.GetVdbById(ctx, vdbId).Execute()
 	})
 
 	if diags != nil {
-		ErrorLog.Printf("Error reading the VDB, removing from state.")
-		d.SetId("")
-		return diag.Errorf("[NOT OK] Error in polling vdb")
+		_, diags, _ := PollForObjectDeletion(func() (interface{}, *http.Response, error) {
+			return client.VDBsApi.GetVdbById(ctx, vdbId).Execute()
+		})
+		// This would imply error in poll for deletion so we just log and exit.
+		if diags != nil{
+			ErrorLog.Printf("[NOT OK] Error in polling vdb")
```

Comment:
Update the error message to have more information. In which poll? what is the error? etc